src/mono/mono.proj

```diff
@@ -580,6 +583,33 @@
       <_MonoSTRIPOption>STRIP="$(_MonoTuple)-strip"</_MonoSTRIPOption>
     </PropertyGroup>
 
+    <!-- x64 illumos cross build options -->
+    <PropertyGroup Condition="'$(TargetsIllumos)' == 'true' and '$(MonoCrossDir)' != ''">
+      <_MonoTuple>x86_64-sun-solaris2.10</_MonoTuple>
+      <_ToolPrefix>$(MonoCrossDir)/bin/x86_64-illumos</_ToolPrefix>
+
+      <_MonoRANLIBOption>RANLIB="$(_ToolPrefix)-ranlib"</_MonoRANLIBOption>
+      <_MonoAROption>AR="$(_ToolPrefix)-ar"</_MonoAROption>
+      <_MonoASOption>AS="$(_ToolPrefix)-as"</_MonoASOption>
+      <_MonoLDOption>LD="$(_ToolPrefix)-ld"</_MonoLDOption>
+      <_MonoSTRIPOption>STRIP="$(_ToolPrefix)-strip"</_MonoSTRIPOption>
+
+      <_MonoCCOption>CC="$(_ToolPrefix)-$(MonoCCompiler)"</_MonoCCOption>
+      <_MonoCXXOption Condition="'$(MonoCCompiler)' == 'gcc'">CXX="$(_ToolPrefix)-g++"</_MonoCXXOption>
```

Comment:
We should move this logic that turns gcc into g++ into a separate `MonoCxxCompiler` property, and probably also do it for clang->clang++.

This matches the logic in https://github.com/dotnet/runtime/blob/bab2bc4bfaaae52860e8f6f8ef1770c1030e70f1/eng/common/native/find-native-compiler.sh#L31-L36

Reply:
Defined `MonoCxxCompiler` separately. Currently only used for illumos cross compiler toolchain. On some platforms, there is no separate clang++ or g++ and the driver program `clang` or `gcc` is used, so it can vary by platform.